lib/smart_proxy_reports/friendly_message.rb

```diff
@@ -0,0 +1,58 @@
+class FriendlyMessage
+  def initialize(result)
+    @result = result
+    @result_tree = @result["result"]
+    @task_tree = @result["task"]
+    @module_args_tree = @result_tree.dig("invocation", "module_args")
+  end
+
+  def generate_message
+    msg = "N/A"
+    return msg if @task_tree.nil? || @task_tree["action"].nil?
+    return @result_tree["msg"] if @result["failed"]
+
+    case @task_tree["action"]
+    when "ansible.builtin.package", "package" then msg = package_message
+    when "ansible.builtin.template", "template" then msg = template_message
+    when "ansible.builtin.service", "service" then msg = service_message
+    when "ansible.builtin.group", "group" then msg = group_message
+    when "ansible.builtin.user", "user" then msg = user_message
+    when "ansible.builtin.cron", "cron" then msg = cron_message
+    when "ansible.builtin.copy", "copy" then msg = copy_message
+    when "ansible.builtin.command", "ansible.builtin.shell", "command", "shell" then msg = @result_tree["stdout_lines"]
+    end
+    msg
+  rescue StandardError => e
+    logger.debug "Unable to parse result (#{e.message}): #{@result.inspect}"
+    msg
+  end
+
+  def package_message
+    detail = @result_tree["results"] || @result_tree["msg"] || "No details"
+    return "Package(s) #{@module_args_tree["name"].join(",")} are #{@module_args_tree["state"]}: #{detail}"
```

Comment:
The last statement in Ruby is the return value by default.
```suggestion
    "Package(s) #{@module_args_tree["name"].join(",")} are #{@module_args_tree["state"]}: #{detail}"
```